hs_core/models.py

```diff
@@ -3437,15 +3437,15 @@ def validate_new_path(cls, new_path):
     def get(cls, resource, file, folder=''):
         """Get a ResourceFile record via its short path."""
         if resource.resource_federation_path:
-            return ResourceFile.objects.get(object_id=resource.id,
-                                            fed_resource_file=get_resource_file_path(resource,
-                                                                                     file,
-                                                                                     folder))
+            return ResourceFile.objects.filter(object_id=resource.id,
+                                               fed_resource_file=get_resource_file_path(resource,
+                                                                                        file,
+                                                                                        folder)).first()
         else:
-            return ResourceFile.objects.get(object_id=resource.id,
-                                            resource_file=get_resource_file_path(resource,
-                                                                                 file,
-                                                                                 folder))
+            return ResourceFile.objects.filter(object_id=resource.id,
+                                               resource_file=get_resource_file_path(resource,
+                                                                                    file,
+                                                                                    folder)).first()
 
```

Comment:
You also have to check that we found a file record match. If not then we need to raise ObjectDoesNotExist.

Reply:
Thanks I'll find those